src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -683,9 +683,11 @@ func (s *service) fetchTagged(ctx context.Context, db storage.Database, req *rpc
 		return nil, err
 	}
 
-	// Step 2: If fetching data read the results of the asynchronuous block readers.
+	// Step 2: If fetching data read the results of the asynchronous block readers.
 	if fetchData {
-		s.fetchReadResults(ctx, response, nsID, encodedDataResults)
+		if err := s.fetchReadResults(ctx, response, nsID, callStart, encodedDataResults); err != nil {
+			return nil, err
+		}
 	}
 
 	s.metrics.fetchTagged.ReportSuccess(s.nowFn().Sub(callStart))
```

Comment:
nit: maybe worth pulling out the `s.metrics.fetchTagged.ReportError(...)` that happens in `fetchReadEncoded` and `fetchReadResults`? That way, the metric reporting all happens at this same method layer in `fetchTagged`?